bazel/linux/README.md

```diff
@@ -2,33 +2,41 @@
+This has two other dependencies: a root filesystem image and executable user-mode linux kernel image (compatible with the kernel version used to compile the module to be tested).
+When executed using *bazel test*, it will launch the user-mode linux image using the provided rootfs image and exposing the module to be tested through hostfs. For more info see [run_um_kunit_tests.sh](https://github.com/enfabrica/enkit/blob/master/bazel/linux/run_um_kunit_tests.sh).
+
+\* *These are repository rules, they are usually added to a project WORKSPACE file, and can then be referenced using the @rule_name syntax.*
 
 # Workflow
 ## Building a kernel module
 1. Generate a .tar.gz using either [kbuild](https://github.com/enfabrica/enkit/tree/master/kbuild) or the [generate_custom_archive.sh](https://github.com/enfabrica/enkit/tree/master/kbuild/utils) script
 2. Make the .tar.gz available through some https mirror at $URL
-3. Define a `kernel_version(name="my-kernel-version", url="$URL", ...)` rule in your repository WORKSPACE file
-4. Add a `kernel_module(name="my-kernel-module", kernel="my-kernel-version", ...)` rule inside your BUILD.bazel file in your kernel module directory
+3. Add a `kernel_tree_version(name="my-kernel-tree", url="$URL", ...)` rule in your repository WORKSPACE file
+4. Add a `kernel_module(name="my_module", kernel="my-kernel-tree", ...)` rule inside your BUILD.bazel file in your kernel module directory
 
 ## Testing a kernel module
-1. Build a kernel module following the instructions above
-2. Add an `http_file(name="rootfs-img", ...)` rule in your repository WORKSPACE file (same for the executable kernel image)
+1. Declare how to build the test kernel module following the instructions above
+2. Add a `rootfs_version(name="my-rootfs", ulr="$URL", ...)` rule in your repository WORKSPACE file
+3. Add a `kernel_image_version(name="my-kernel-image", ulr="$URL", ...)` rule in your repository WORKSPACE file
    * Check out the *generate_custom_archive.sh* [instructions](https://github.com/enfabrica/enkit/blob/master/kbuild/utils/README.md) if you don't know how to generate a suitable  user-mode linux executable image
-4. Add a `kernel_test(name="my-kernel-module-test", module=":my-kernel-module", kernel_image="@kernel-img//file", rootfs_image="@rootfs-img//file")` rule inside your BUILD.bazel file in your kernel module directory
-5. Test the kernel running `bazel test :my-kernel-module-test`
+4. Add a `kernel_test(name="my_module_test", module=":my_module", kernel_image="@my-kernel-image", rootfs_image="@my-rootfs")` rule inside your BUILD.bazel file in your kernel module directory
+5. Test the kernel running `bazel test :my_module_test`
+   * Use `--test_output=all` if you want to see the user-mode linux and kunit output
```

Comment:
This is all very good documentation! As a side note, I think we should have an example checked in in the public repo.

Reply:
Agreed. I'll come up with an E2E example for running a test.